# Bài 5.5: Kiểm định Hotelling $T^2$

Notebook này tái lập kiểm định $H_0:\mu^T=[0.55, 0.60]$ ở mức ý nghĩa $\alpha=0.05$.

In [5]:
from pathlib import Path
from statistics import NormalDist

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from IPython.display import display

pd.set_option("display.precision", 4)
plt.rcParams.update({
    "figure.dpi": 120,
    "axes.grid": True,
    "grid.alpha": 0.25,
})

## 1. Dữ kiện từ Ví dụ 5.3

Sử dụng các đại lượng $\bar{x}$, $S^{-1}$ và cỡ mẫu trong lời giải.

In [6]:
n = 42
p = 2
alpha = 0.05

xbar = np.array([0.564, 0.603])
mu0 = np.array([0.55, 0.60])
S_inv = np.array(
    [
        [203.018, -163.391],
        [-163.391, 200.228],
    ]
)

display(pd.DataFrame(xbar, index=["Variable 1", "Variable 2"], columns=["xbar"]))
display(pd.DataFrame(S_inv, index=["Variable 1", "Variable 2"], columns=["Variable 1", "Variable 2"]))

,xbar
Variable 1,0.564
Variable 2,0.603


,Variable 1,Variable 2
Variable 1,203.018,-163.391
Variable 2,-163.391,200.228


## 2. Tính thống kê kiểm định

In [7]:
d = xbar - mu0
T2 = n * d.T @ S_inv @ d

display(pd.DataFrame(d, index=["Variable 1", "Variable 2"], columns=["d = xbar - mu0"]))
T2

,d = xbar - mu0
Variable 1,0.014
Variable 2,0.003


np.float64(1.1704869119999803)

## 3. Giá trị tới hạn và kết luận

Theo công thức kiểm định Hotelling,


$$
T^2 > \frac{(n-1)p}{n-p}F_{p,n-p}(\alpha)
$$


thì bác bỏ $H_0$. Trong lời giải dùng $F_{2,40}(0.05)$ sao cho giá trị tới hạn xấp xỉ $6.625$.

In [8]:
critical_value = 6.625
decision = "Reject H0" if T2 > critical_value else "Do not reject H0"

pd.DataFrame(
    {
        "T^2": [T2],
        "critical value": [critical_value],
        "decision": [decision],
    }
)

,T^2,critical value,decision
0,1.1705,6.625,Do not reject H0


Vì $T^2 \approx 1.170 < 6.625$, ta không bác bỏ $H_0$. Kết quả này nhất quán với elip tin cậy 95% trong Hình 5.1 vì điểm $[0.55, 0.60]^T$ nằm trong miền tin cậy.